cpp/include/raft/neighbors/detail/ivf_flat_search-inl.cuh

```diff
@@ -42,25 +43,35 @@ void search_impl(raft::resources const& handle,
                  uint32_t queries_offset,
                  uint32_t k,
                  uint32_t n_probes,
+                 uint32_t max_samples,
                  bool select_min,
                  IdxT* neighbors,
                  AccT* distances,
                  rmm::mr::device_memory_resource* search_mr,
                  IvfSampleFilterT sample_filter)
 {
   auto stream = resource::get_cuda_stream(handle);
+
+  std::size_t n_queries_probes = std::size_t(n_queries) * std::size_t(n_probes);
+
   // The norm of query
   rmm::device_uvector<float> query_norm_dev(n_queries, stream, search_mr);
   // The distance value of cluster(list) and queries
   rmm::device_uvector<float> distance_buffer_dev(n_queries * index.n_lists(), stream, search_mr);
   // The topk distance value of cluster(list) and queries
-  rmm::device_uvector<float> coarse_distances_dev(n_queries * n_probes, stream, search_mr);
+  rmm::device_uvector<float> coarse_distances_dev(n_queries_probes, stream, search_mr);
   // The topk  index of cluster(list) and queries
-  rmm::device_uvector<uint32_t> coarse_indices_dev(n_queries * n_probes, stream, search_mr);
+  rmm::device_uvector<uint32_t> coarse_indices_dev(n_queries_probes, stream, search_mr);
+
+  // Optional structures if postprocessing is required
   // The topk distance value of candidate vectors from each cluster(list)
-  rmm::device_uvector<AccT> refined_distances_dev(n_queries * n_probes * k, stream, search_mr);
+  rmm::device_uvector<AccT> refined_distances_dev(0, stream, search_mr);
```

Comment:
Here these arrays just store temporary output values after IVF-Flat search. Since writing the IVF-Flat code, we have added a [refinement](https://github.com/rapidsai/raft/blob/c95bf6aecb48f8eb751e4b706b727e48eaf9757f/cpp/include/raft/neighbors/refine-inl.cuh#L70) operation which is applicable after vector search with lossy compression (like IVF-PQ), and it is unrelated to what we do here. That makes the variable names `refined_` somewhat confusing. May I ask you to rename these arrays to `distances_tmp_dev` or something similar without the `refined` term?